### ISSUE TITLE
Suggest when I am clicked should work on stage

### ISSUE THREAD

**Issue by brianharvey** [jmoenig/Snap--Build-Your-Own-Blocks#1300]
It should mean that the user clicked on a place on the stage not covered by a sprite.  Not super important, it just would have been convenient for a project.

**Comment by jmoenig** [comment]
wait! doesn't that work for you? I'm using it all the time:
http://snap.berkeley.edu/snapsource/snap.html#present:Username=jens&ProjectName=Towers%20of%20Hanoi

**Comment by jmoenig** [comment]
Can somebody confirm whether this is working (as it is for me)? Thanks!

**Comment by brianharvey** [comment]
Yes, sorry, my experiment failed for a different reason, namely a bug or maybe misfeature in NEXT COSTUME: If the costume (in this case the background) is 0, it doesn't switch to 1.  IIRC what we decided for BYOB was that NEXT COSTUME would never switch *into* costume 0 (wrapping to 1), but that it *would* switch from 0 to 1.  And I think that's still what it should do.  Sorry about misreporting the problem.

**Comment by jmoenig** [comment]
BYOB didn't have the the special null costume that is a LOGO turtle-ish pen in Snap, BYOB didn't let you delete the last costume, you always had to have at least one costume per sprite. I want to keep the Turtle costume a special case, because it really mean "no" costume to me. If you want to have a turtle costume in the loop, you can simply make your own. I'm also not convinced about the asymmetrical proposed rule to loop *out* of the zero costume but never *in* again.

**Comment by brianharvey** [comment]
You're right, my memory is going.  But still, it's the wrapping that has to be thought about specially, from costume N to costume 1.  If you're at costume 0, and you say "next costume," you're expecting *something* to happen, and the next number after 0 is 1.

Remember, the situation in which it came up for me was on the stage, not in a sprite, and background 0 isn't so obviously special.

This is a case where I think we should follow the Scratch principle that anything you can say should mean *something.*

**Comment by cycomachead** [comment]
I think many students consider the to be a costume. However, since it can't be deleted, the current behavior does make sense if you're switching through costumes in a program.

**Comment by ToonTalk** [comment]
The following is like next costume but goes from 0 to 1

![image](https://user-images.githubusercontent.com/2138696/36714233-ecd83c9e-1bc3-11e8-898b-127325fd4a01.png)

Maybe Brian is right and next costume should go 0 to 1. Except the above goes from last costume to 0 instead of 1.

**Comment by jguille2** [comment]
Yes Ken,

And this [switch to costume((costume#) + 1)] goes from the last costume to the _turtle_. And it's a good solution for somebody that wants the _turtle_ into the _costume loop_.

But we want the _turtle_ out of the loop! Because we can not change it...

Then, if I want the turtle into the loop, I can apply two solutions: to use something like your block (to put the turtle inside the loop) or to create a new turtle custom (it will not have pen turtle properties).

Other question is the [next costume] action over the turtle. For me, current behavior is right and consistent (turtle is not a costume, so [next costume] does nothing). Anyway, if [next costume] goes to the first costume (is the next (closer) costume) it wold be also fine. But the important tip is **that _turtle_ must be out of the loop**.

Joan

**Comment by ToonTalk** [comment]
I agree about the loop.

I think the unclear design question is the "Other question" -- what next costume does over the turtle. I see both arguments in this issue and am not sure what's right. But the current situation is not very consistent regarding whether the turtle is a costume. E.g. "switch to costume" treats it as such while "next costume" doesn't.

**Comment by brianharvey** [comment]
I think the abstract part of this discussion, the part about whether the turtle costume is a costume, is just silly.  Of course it's a costume!  Go listen to users talking.  You'll hear approximately ℵ<sub><small>0</small></sub> instances of "the turtle costume" for every instance of "The sprite doesn't have a costume; it's being a turtle instead."

So that leaves the question of how NEXT COSTUME should behave.  I believe all of us agree that when a sprite is wearing its highest-numbered costume, NEXT COSTUME should go to costume 1, not costume 0.

So all that's left to argue about is what NEXT COSTUME should do when the sprite is wearing its turtle costume. There are only three possibilities:  It can give an error message, which would be useless; it can do nothing at all, which is far worse than useless; or it can do something useful.

So there's nothing left to argue about except to take bets on how long it will take for Jens to see the light about this.  :P
